src/main/java/core/basesyntax/reportservices/ReportCreatorImpl.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.reportservices;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class ReportCreatorImpl implements ReportCreator {
+    @Override
+    public List<String> report(Map<String, Integer> transactions) {
+        List<String> report = new ArrayList<>();
```

Comment:
Consider using a more specific List implementation that is optimized for the operation. Since the report size is known after reading the transactions map, initializing the ArrayList with the expected size could improve performance.